scalafix-rules/src/main/scala/scalafix/internal/rule/ExplicitResultTypes.scala

```diff
@@ -92,35 +92,32 @@ final class ExplicitResultTypes(
     }
   }
 
-  override def fix(implicit ctx: SemanticDocument): Patch = {
+  override def fix(implicit ctx: SemanticDocument): Patch =
     try unsafeFix()
     catch {
-      case _: CompilerException =>
-        shutdownCompiler()
```

Comment:
If the compiler crashes because usually it cannot type a part of the AST, it will crash again even after a retry. 
Actually we do that on the file level, which means if one explicit type makes the compiler crash, we don't add any types in the entire file. Should we continue to do that ?

Reply:
> The compiler crashes sometimes because it cannot type a part of the AST. If it crashes once, it will crash again even after a retry.

Is that the only case? https://github.com/scalacenter/scalafix/commit/ba74db38dde7fcdd4971b56b10ec94cb6fef7092 is not very verbose, I am not sure... In any case, if you reproduced the issue, it would be good to document the cases where the compiler crashes and how it behaves via tests.

> Actually we do that on the file level, which means if one explicit type makes the compiler crash, we don't add any types to the entire file. Should we continue to do that ?

Catching the exception at a more granular seems like a good idea indeed. Maybe this wasn't done for performance reason because some files would crash the compiler over and over again?